lib/node_modules/@stdlib/math/base/special/normhermite-polynomial/lib/main.js

```diff
@@ -18,20 +18,70 @@
 
 'use strict';
 
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var isint = require( '@stdlib/math/base/assert/is-integer' );
+
+
 // MAIN //
 
 /**
 * Evaluates a normalized Hermite polynomial.
 *
-* @param {<TODO: type>} TODO - TODO: param desc
-* @returns {<TODO: type>} TODO: desc
+* @param {integer} n - non-negative polynomial degree
+* @param {number} x - evaluation point
+* @returns {number} function value
+*
+* @example
+* var v = normhermitepoly( 1, 0.5 );
+* // returns 0.5
+*
+* @example
+* var v = normhermitepoly( -1, 0.5 );
+* // returns NaN
 *
 * @example
-* var v = normhermitepoly( 1, 1.0 );
-* // returns TODO
+* var v = normhermitepoly( 0, 0.5 );
+* // returns 1.0
+*
+* @example
+* var v = normhermitepoly( 2, 0.5 );
+* // returns -0.75
 */
-function normhermitepoly() {
-	// TODO: implementation
+function normhermitepoly( n, x ) {
+	var y1;
+	var y2;
+	var y3;
+	var i;
+
+	if ( isnan( n ) || isnan( x ) ) {
```

Comment:
You can consolidate your branching into a single `if` statement. Meaning, we don't need 3 separate `if` statements to return the same value. Here, we can consolidate all the conditions into a single `if` statement condition.

```javascript
if ( isnan( x ) || isnan ( n ) || n < 0 || !isint( n ) ) {
    ...
}
```

Note that I also put the negative check before the integer check as `isint` is likely to be slightly more expensive, even if inlined.

Reply:
Nice call on your part.